src/engine/guidance/post_processing.cpp

```diff
@@ -198,6 +198,7 @@ std::vector<std::vector<PathData>> postProcess(std::vector<std::vector<PathData>
     // silence silent turns for good
     for (auto &path_data : leg_data)
```

Comment:
Can you explain to me what you do here?

The fix should be not set `data.duration_until_turn = 0;` Or merge the duration with the previous segment.